src/controllers/user.controller.ts

```diff
@@ -87,4 +130,31 @@ export class UserController {
   ): Promise<Product[]> {
     return this.recommender.getProductRecommendations(userId);
   }
+
+  @post('/users/login', {
+    responses: {
+      '200': {
+        description: 'Token',
+        content: {
+          'application/json': {
+            schema: {
+              type: 'object',
+              properties: {
+                token: {
+                  type: 'string',
+                },
+              },
+            },
+          },
+        },
+      },
+    },
+  })
+  async login(
+    @requestBody() credentials: Credentials,
+  ): Promise<{token: string}> {
+    let res = {token: ''};
+    res.token = await this.userRepository.login(credentials);
```

Comment:
It's a bit awkward that the `login` method is implemented by `UserRepository`. Should we move it into the controller itself?

Reply:
The code can be simplified as:
```ts
const token = await this.userRepository.login(credentials);
return {
  token,
};
```

Reply:
@raymondfeng There are different opinions regarding it, see https://github.com/strongloop/loopback4-example-shopping/pull/26#discussion_r246745742
At first it was implemented in the controller, but then I hesitate, the good thing of having that in the repository is we don't make http calls to get the access token when test.